core/src/test/java/org/elasticsearch/index/query/TermsQueryBuilderTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import com.google.common.collect.Lists;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.indices.cache.query.terms.TermsLookup;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.hamcrest.Matchers.*;
+
+public class TermsQueryBuilderTest extends BaseQueryTestCase<TermsQueryBuilder> {
+
+    @Override
+    protected TermsQueryBuilder doCreateTestQueryBuilder() {
+        Tuple<String, Object> fieldAndValue = getRandomFieldNameAndValue();
+
+        // terms query or lookup query
+        TermsQueryBuilder query;
+        if (randomBoolean()) {
+            // make between 0 and 5 number of the same value
+            Object[] values = new Object[randomInt(5)];
+            for (int i = 0; i < values.length; i++) {
+                values[i] = fieldAndValue.v2();
```

Comment:
I understand the that the way the we test the created lucene query at the moment is not affected, however since we already have the test for `expected list of terms` hat checks the terms at least for string fields, I think it would be really good to have this tests. Changing this shouldn't be too hard, and also the random query we create here would feel more "natural". Please correct me if this change is hard to do, otherwise it would be really great to have random values here.